index.js

```diff
@@ -0,0 +1,75 @@
+const https = require('https');
+const { parseString } = require('xml2js');
+
+async function getAllUrlsFromSitemapIndex(siteUrl) {
+  try {
+    const sitemapIndexUrl = `${siteUrl}/sitemap.xml`;
+    const sitemapIndexResponse = await fetchContent(sitemapIndexUrl);
+
+    const sitemapIndexXml = sitemapIndexResponse.data;
+    const sitemapIndexResult = await parseXml(sitemapIndexXml);
+    const sitemapUrls = sitemapIndexResult.sitemapindex.sitemap.map(sitemap => sitemap.loc[0]);
+    const fetchSitemapPromises = sitemapUrls.map(url => fetchContent(url));
+    const sitemapResponses = await Promise.all(fetchSitemapPromises);
+
+    let allUrls = [];
+    for (const response of sitemapResponses) {
+      const sitemapResult = await parseXml(response.data);
+
+      if (sitemapResult.urlset && sitemapResult.urlset.url) {
+        const sitemapUrls = sitemapResult.urlset.url.map(urlObj => urlObj.loc[0]);
+        allUrls = allUrls.concat(sitemapUrls);
+      }
+    }
+
+    return allUrls;
+  } catch (error) {
+    throw error;
+  }
+}
+
+function fetchContent(url) {
+  return new Promise((resolve, reject) => {
+    https.get(url, (response) => {
+      let content = '';
+
+      response.on('data', (chunk) => {
+        content += chunk;
+      });
+
+      response.on('end', () => {
+        if (response.statusCode === 200) {
+          resolve({ data: content, url });
+        } else {
+          reject(new Error(`Failed to fetch content from ${url}. Status code: ${response.statusCode}`));
+        }
+      });
+    }).on('error', (error) => {
+      reject(error);
+    });
+  });
+}
+
+function parseXml(xml) {
+  return new Promise((resolve, reject) => {
+    parseString(xml, (err, result) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+// Example 
+const siteUrl = 'https://www.budgetgolf.com';
```

Comment:
The user can enter anything in URL. So, We need to validate it before starting it. If it is valid url and it found sitemap.xml then only it should go to next process.